gstreamer-sdp/src/s_d_p_media.rs

```diff
@@ -190,249 +200,225 @@ impl SDPMedia {
-                _ => Err(result),
+                _ => Err(()),
             }
         }
     }
 
-    pub fn remove_attribute(&mut self, idx: u32) -> Result<_, ()> {
+    pub fn remove_attribute(&mut self, idx: u32) -> Result<(), ()> {
         let result = unsafe {
             from_glib(ffi::gst_sdp_media_remove_attribute(self.to_glib_none_mut().0, idx))
         };
         match result {
-            SDPResult::ok => Some(_),
+            SDPResult::ok => Some(()),
             _ => Err(()),
             }
         }
     }
 
-    pub fn remove_bandwidth(&mut self, idx: u32) -> Result<_, ()> {
+    pub fn remove_bandwidth(&mut self, idx: u32) -> Result<(), ()> {
         let result = unsafe {
             from_glib(ffi::gst_sdp_media_remove_bandwidth(self.to_glib_none_mut().0, idx))
         };
         match result {
-            SDPResult::ok => Some(_),
+            SDPResult::ok => Some(()),
             _ => Err(()),
             }
         }
     }
 
-    pub fn remove_connection(&mut self, idx: u32) -> Result<_, ()> {
+    pub fn remove_connection(&mut self, idx: u32) -> Result<(), ()> {
         let result = unsafe {
             from_glib(ffi::gst_sdp_media_remove_connection(self.to_glib_none_mut().0, idx))
         };
         match result {
-            SDPResult::ok => Some(_),
+            SDPResult::ok => Some(()),
             _ => Err(()),
             }
         }
     }
 
-    pub fn remove_format(&mut self, idx: u32) -> Result<_, ()> {
+    pub fn remove_format(&mut self, idx: u32) -> Result<(), ()> {
         let result = unsafe {
             from_glib(ffi::gst_sdp_media_remove_format(self.to_glib_none_mut().0, idx))
         };
         match result {
-            SDPResult::ok => Some(_),
+            SDPResult::ok => Some(()),
             _ => Err(()),
             }
         }
     }
 
-    pub fn replace_attribute(&mut self, idx: u32, attr: &mut SDPAttribute) -> Result<_, ()> {
+    pub fn replace_attribute(&mut self, idx: u32, attr: SDPAttribute) -> Result<(), ()> {
         let result = unsafe {
-            from_glib(ffi::gst_sdp_media_replace_attribute(self.to_glib_none_mut().0, idx, attr.to_glib_none_mut().0))
+            from_glib(ffi::gst_sdp_media_replace_attribute(self.to_glib_none_mut().0, idx, attr.to_glib_full()))
         };
+        mem::forget(attr);
         match result {
-            SDPResult::ok => Some(_),
+            SDPResult::ok => Some(()),
             _ => Err(()),
             }
         }
     }
 
-    pub fn replace_bandwidth(&mut self, idx: u32, bw: &mut SDPBandwidth) -> Result<_, ()> {
+    pub fn replace_bandwidth(&mut self, idx: u32, bw: SDPBandwidth) -> Result<(), ()> {
         let result = unsafe {
-            from_glib(ffi::gst_sdp_media_replace_bandwidth(self.to_glib_none_mut().0, idx, bw.to_glib_none_mut().0))
+            from_glib(ffi::gst_sdp_media_replace_bandwidth(self.to_glib_none_mut().0, idx, bw.to_glib_full()))
```

Comment:
Also for all these types you did not implement the GLib translation traits so this won't work. But you don't need the traits here, just some pointer casting for these types